relays/ethereum/src/ethereum_client.rs

```diff
@@ -120,13 +121,35 @@ impl EthereumRpc for EthereumRpcClient {
 	}
 
 	async fn header_by_hash(&self, hash: H256) -> Result<Header> {
-		let header = Ethereum::get_block_by_hash(&self.client, hash).await?;
+		let get_full_tx_objects = false;
+		let header = Ethereum::get_block_by_hash(&self.client, hash, get_full_tx_objects).await?;
 		match header.number.is_some() && header.hash.is_some() && header.logs_bloom.is_some() {
 			true => Ok(header),
 			false => Err(RpcError::Ethereum(EthereumNodeError::IncompleteHeader)),
 		}
 	}
 
+	async fn header_by_hash_with_transactions(&self, hash: H256) -> Result<HeaderWithTransactions> {
+		let get_full_tx_objects = true;
+		let header = Ethereum::get_block_by_hash_with_transactions(&self.client, hash, get_full_tx_objects).await?;
+
+		let is_complete_header = header.number.is_some() && header.hash.is_some() && header.logs_bloom.is_some();
+		if !is_complete_header {
+			return Err(RpcError::Ethereum(EthereumNodeError::IncompleteHeader));
```

Comment:
This happens if the header/block is still pending?

Reply:
I haven't found this information in [web3](https://docs.rs/web3/0.13.0/web3/types/struct.Block.html) documentation. But yeah - I think this could happen when block is pending.